### ISSUE TITLE
Unstable inference channel mean rmse

### ISSUE THREAD

**Issue by suryadheeshjith** [m2lines/Samudra#65]
<img width="450" alt="Image" src="https://github.com/user-attachments/assets/d3a7fdb8-fc1f-4796-b284-565837e81bee" />

Possible suspects:
- Stepper outside forward function and batch norm. Linked to [this notebook.](https://github.com/suryadheeshjith/Ocean_Emulator/blob/master/notebooks/batchnorm_test.ipynb) (Related #51
)
- Implement an EMA on model weights

**Comment by jder** [comment]
@suryadheeshjith you mentioned in slack you're working on this. Do you think it would make sense for me to investigate the stepper mechanics linked here or are you already doing that?

**Comment by suryadheeshjith** [comment]
I just solved this. This was mainly due to the use of batch normalization in the model. I have tried with instance / layer normalization and it seems to have fixed this issue. EMA helped with stabilizing them as well.

![Image](https://github.com/user-attachments/assets/65d04552-08fa-4537-8542-eba6b446b572)

**Comment by jder** [comment]
Ok, great. So I could still pick up the stepper work but just as a clean up rather than for this?

**Comment by suryadheeshjith** [comment]
Yes!
